R/coerce_to_messydate.R

```diff
@@ -105,10 +120,76 @@ standardise_unspecifieds <- function(dates) {
   dates <- stringr::str_replace_all(dates, "-00-|-0-|-0$|-00$", "-XX-")
   dates <- stringr::str_replace_all(dates, "\\?\\?\\?\\?", "XXXX")
   dates <- stringr::str_replace_all(dates, "-\\?\\?", "-XX")
+  dates <- stringr::str_replace_all(dates, "-XX$", "")
+  dates <- ifelse(stringr::str_detect(dates, "^[:digit:]{4}\\~$"),
+                  paste0("~", stringr::str_remove(dates, "\\~")), dates)
+  dates
+}
+
+standardise_date_input <- function(dates) {
+  dates <- ifelse(stringr::str_detect(dates, "(bc|BC|Bc|bC)"),
+                  as_bc_dates(dates), dates)
+  dates <- ifelse(stringr::str_detect(dates, "(ad|AD|Ad|aD)"),
+                  as_ac_dates(dates), dates)
+  dates <- gsub(" ", "", dates)
   dates
 }
 
 standardise_widths <- function(dates) {
+  dates <- add_zero_padding(dates)
+  dates <- ifelse(stringr::str_detect(dates, "([:digit:]{1})\\.\\.([:digit:]{1})|([:digit:]{1})\\.\\.-"),
+                  add_zero_range(dates), dates)
+  dates <- ifelse(stringr::str_detect(dates, "\\,"),
+                  add_zero_set(dates), dates)
+  dates <- ifelse(stringr::str_detect(dates, "\\,"),
+                  paste0("{", dates, "}"), dates)
+  dates <- stringr::str_replace_all(dates, "-([:digit:])$", "-0\\1")
+  dates <- stringr::str_replace_all(dates, "^([:digit:])-", "0\\1-")
+  dates
+}
+
+extract_from_text <- function(v) {
```

Comment:
This doesn't work with dates like "twenty second day of November 2022". I'll modify it a bit later

Reply:
Thank you noticing and fixing all these issues @jaeltan !